packages/api/src/command/medical/cq-directory/rebuild-cq-directory.ts

```diff
@@ -0,0 +1,105 @@
+import { Carequality } from "@metriport/carequality-sdk/client/carequality";
+import { sleep } from "@metriport/core/util/sleep";
+import { QueryTypes, Sequelize } from "sequelize";
+import { Config } from "../../../shared/config";
+import { capture } from "../../../shared/notifications";
+import { bulkInsertCQDirectoryEntries } from "./create-cq-directory-entry";
+import { parseCQDirectoryEntries } from "./parse-cq-directory-entry";
+
+const BATCH_SIZE = 1000;
+const cqDirectoryEntryTemp = `cq_directory_entry_temp`;
+const cqDirectoryEntry = `cq_directory_entry`;
+const cqDirectoryEntryBackup = `cq_directory_entry_backup`;
+
+const sqlDBCreds = Config.getDBCreds();
+const dbCreds = JSON.parse(sqlDBCreds);
+
+const sequelize = new Sequelize(dbCreds.dbname, dbCreds.username, dbCreds.password, {
+  host: dbCreds.host,
+  port: dbCreds.port,
+  dialect: dbCreds.engine,
+});
+
+export const rebuildCQDirectory = async (failGracefully = false): Promise<void> => {
+  let currentPosition = 0;
+  let isDone = false;
+
+  try {
+    await createTempCQDirectoryTable();
+    while (!isDone) {
+      try {
+        const apiKey = Config.getCQApiKey();
+        const cq = new Carequality(apiKey);
+        const orgs = await cq.listOrganizations({ start: currentPosition });
+        if (orgs.length < BATCH_SIZE) isDone = true; // if CQ directory returns less than BATCH_SIZE number of orgs, that means we've hit the end
+        currentPosition += BATCH_SIZE;
+        const parsedOrgs = parseCQDirectoryEntries(orgs);
+        await bulkInsertCQDirectoryEntries(sequelize, parsedOrgs);
+        console.log(`Added ${currentPosition} CQ directory entries...`);
+        await sleep(750);
```

Comment:
Can we have `750` as a duration on the top of the file to make it easier to customize if needed, please?

Reply:
> as a duration

😄